etna/records/templatetags/records_tags.py

```diff
@@ -62,7 +62,9 @@ def record_url(
         ):
             return f"https://discovery.nationalarchives.gov.uk/details/r/{iaid}"
         try:
-            return reverse("details-page-machine-readable", kwargs={"iaid": iaid})
+            return reverse(
+                "details-page-machine-readable", kwargs={"metadata_id": iaid}
+            )
         except NoReverseMatch:
             pass
     if ref:
```

Comment:
There's a similar issue in this module: Again, we are taking the `.iaid` property value from Records (or `["iaid"]` key from search results). This value won't be available for the new records we hope to support. We need to be using something other than `iaid` from the data here

Reply:
Same as mentioned previously: The current api response and attribute have the same name i.e. iaid until now, no other source is formalised.

Further, I mentioned about `primaryIdentifier` we need to know further about this if at all this this the generic one. 

@ababic , perhaps to treat this as a separate issue and lets go with the current status of using metadataId and Iaid. Thoughts?

Reply:
If we are in limbo with terminology, then we should make the **fewest number of changes we possibly can** for now, then bring everything in-line once we have more certainty. Specifically, I would recommend changing nothing but the API client logic, so that we're using `metadata_id` as the parameter name instead of `iaid`.

Reply:
>  I would recommend changing nothing but the API client logic, so that we're using `metadata_id` as the parameter name instead of `iaid`. - Everything else remains the same. 

In this case, the `Record.api.fetch(iaid=iaid)` where the keyword input param `iaid` is confusing  whereas the API actually uses `metadataId` internally. 

I had initially thought about removing the keyword input param when calling `api.fetch` so it will become `Record.api.fetch(iaid)`, but then I wanted it to be explicit `Record.api.fetch(metadata_id=iaid)` as we know what `metadata_id` is about as mentioned in the parameter comments.  I think something like a object for multiple formats.

If or when we know that a record does not hold `iaid` we can address it then. 

This are my thoughts @ababic

Reply:
Thanks @ababic updated with minimal api changes.